Mathlib/Algebra/Order/Sub/Defs.lean

```diff
@@ -0,0 +1,452 @@
+/-
+Copyright (c) 2021 Floris van Doorn. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Floris van Doorn
+-/
+import Mathlib.Algebra.CovariantAndContravariant
+import Mathlib.Algebra.Group.Basic
+import Mathlib.Algebra.Order.Monoid.Lemmas
+import Mathlib.Order.Lattice
+
+/-!
+# Ordered Subtraction
+
+This file proves lemmas relating (truncated) subtraction with an order. We provide a class
+`HasOrderedSub` stating that `a - b ≤ c ↔ a ≤ c + b`.
+
+The subtraction discussed here could both be normal subtraction in an additive group or truncated
+subtraction on a canonically ordered monoid (`ℕ`, `Multiset`, `PartEnat`, `Ennreal`, ...)
```

Comment:
```suggestion
subtraction on a canonically ordered monoid (`ℕ`, `Multiset`, `PartENat`, `ENnReal`, ...)
```
Not entirely sure about `ENnReal`, but I think that's what the current rule says!